Modules/_json.c

```diff
@@ -1572,17 +1573,15 @@ encoder_listencode_dict(PyEncoderObject *s, _PyAccu *acc,
     items = PyMapping_Items(dct);
     if (items == NULL)
         goto bail;
-    if (s->sort_keys && PyList_Sort(items) < 0) {
-        Py_DECREF(items);
-        goto bail;
-    }
+
+    coerced_items = PyList_New(0);
```

Comment:
This seems to introduce a few possible refleaks for `coerced_items`, effectively in each `goto bail` location where coerced_items isn't `NULL`, prior to the `Py_DECREF(coerced_items)`. IMO, the easiest solution would be adding a `Py_XDECREF(coerced_items)` to `bail` since there are multiple paths where this could potentially occur.

There should probably also be the following error condition, in case the list can't be instantiated:
```suggestion
    coerced_items = PyList_New(0);
    if (coerced_items == NULL) {
        goto bail;
    }
```

What do you think @serhiy-storchaka? My experience with the C-API is fairly limited, so I'm not entirely certain about the above.